src/julia.h

```diff
@@ -457,15 +457,6 @@ typedef struct _jl_code_instance_t {
 
     // purity results
     // see also encode_effects() and decode_effects() in `base/compiler/effects.jl`,
-    uint32_t ipo_purity_bits;
```

Comment:
keep the `ipo_purity_bits` name, please. We're not doing any non-IPO optimizations at the moment, but we may in the future, so there's no reason to rename it now.

Reply:
I was keeping the shorter name, due to discussion with @aviatesk. Non-IPO optimizations would presumably be done locally anyways, and not require this global metadata?

Reply:
No, we may need to cache them if we want to allow non-IPO inlining which will be desirable for our float functions once we properly effect model them rather than pretending they're consistent.

Reply:
What is non-interprocedural inlining and why doesn't it require source code?

Reply:
I meant that as a shorthand. In general, you may want to re-run the full optimization pipeline in non-IPO mode, including concrete eval (which needs to know the non-IPO variant of the bits) DCO (which also needs it) and inlining (which needs a non-IPO version of the source as you point out). Perhaps inlining is confusing because it only really make sense if the non-IPO version is inlineable, but the IPO version is not.

Reply:
So are you just saying you want the longer name kept for now instead? It seemed like all the names here refer to the IPO info, so it seemed redundant, but I can switch which name is kept if that is all you meant

Reply:
Yes, keep the name